sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1374,4 +1377,47 @@ class HiveDDLSuite
       assert(e2.message.contains("Hive data source can only be used with tables"))
     }
   }
+
+  test("table partition schema should be ordered") {
+    withTable("t", "t1") {
+      val path = Utils.createTempDir(namePrefix = "t")
+      val path1 = Utils.createTempDir(namePrefix = "t1")
+      try {
+        spark.sql(s"""
+                     |create table t (id long, P1 int, P2 int)
+                     |using parquet
+                     |options (path "$path")
+                     |partitioned by (P1, P2)""".stripMargin)
```

Comment:
this test can pass without your changes right? I think we can just keep the below one.